### DISCUSSION TITLE
Negative and Positive sampling with NeighborLoader

### DISCUSSION THREAD

**Discussion by k-s-b** [pyg-team/pytorch_geometric#7838] [Q&A] [answered]
Hi @rusty1s , 

How to sample positive (one hop away) and negative nodes with `NeighborLoader`? 

I want to perform unsupervised training by keeping the embeddings of positive nodes closer to the center nodes, and push embeddings of negative samples away. In the previous versions, it was possible to do it via `NeighborSampler`, as [here](https://github.com/pyg-team/pytorch_geometric/commit/5223014577bb480c596b0392f32952fd1d1fb6f3). 
I see that it is possible to perform negative sampling with `LinkNeighborLoader` but it samples negative edges rather than nodes.

**Comment by rusty1s** [comment, marked as answer]
I think this is exactly what our `graph_sage_unsup.py` example is doing, so please take a look. I don't think you can sample negative nodes without sampling negative edges?
